pkgs/applications/networking/browsers/webmacs/default.nix

```diff
@@ -0,0 +1,83 @@
+{ pkgs
+, lib
+, python3Packages
+, fetchFromGitHub
+}:
+
+let py3pk = python3Packages; in
+
+pkgs.python3.pkgs.buildPythonApplication rec {
+  name = "webmacs-${version}";
+  version = "0.8";
+
+  src = fetchFromGitHub {
+    owner  = "parkouss";
+    repo   = "webmacs";
+    rev    = "6d92ae8c2bf82cce854f34cb3d762932fe7fdc8b";
+
+    fetchSubmodules = true;
+
+    # The sha256 value was obtained by running:
+    # nix-prefetch-git https://github.com/parkouss/webmacs --fetch-submodules --rev 6d92ae8c2bf82cce854f34cb3d762932fe7fdc8b
+    # where the --rev given to nix-prefetch-git is the same as
+    # the rev variable in this set, at the time this comment was written :-)
+
+    sha256 = "1hzb9341hybgrqcy1w20hshm6xaiby4wbjpjkigf4zq389407368";
+
+    # Careful (at the time of writing, 2019-07-01) forgetting the
+    # --fetch-submodules argument to nix-prefetch-git, does NOT result
+    # in a hash mismatch (as it should, because the hashes of the repo
+    # with/without submodules should not match) and then we get a
+    # build error later on, when the submodule directories turn out to
+    # be empty.
+
+  };
+
+  propagatedBuildInputs = [
+    py3pk.pyqt5_with_qtwebkit
+    py3pk.dateparser
+    py3pk.jinja2
+  ];
+
+  buildInputs = [
+    py3pk.pygments
+    pkgs.qt5.qtbase
+  ];
+
+  doCheck = false;
+
+  checkInputs = [
+    py3pk.pytest
+    py3pk.pytest-xvfb
+    py3pk.pytest-qt
+    pkgs.herbstluftwm
+
+    # The following are listed in test-requirements.txt but appear not
+    # to be needed at present:
+
+    # py3pk.pytest-mock
+    # py3pk.flake8
+  ];
+
+  checkPhase = "pytest";
```

Comment:
I'm surprised this isn't run by default but I could be wrong

Reply:
`pytest` is not the default test runner.